File: VB/ExtendedDrillThroughReport/CategoriesReport.vb
Imports System
Imports System.Drawing
Imports System.Collections
Imports System.ComponentModel
Imports DevExpress.XtraReports.UI
Imports System.Windows.Forms
Imports System.Data
Imports DevExpress.XtraPrinting
Imports DevExpress.XtraPrinting.Preview

Namespace ExtendedDrillThroughReport
	Partial Public Class CategoriesReport
		Inherits DevExpress.XtraReports.UI.XtraReport
		Implements IDrillThroughReportHelper

		#Region "IDrillThroughReportHelper"
		Public Property Viewer() As DocumentViewer Implements IDrillThroughReportHelper.Viewer
		Public Property ParentReport() As XtraReport Implements IDrillThroughReportHelper.ParentReport
		#End Region

		Public Sub New()
			InitializeComponent()
		End Sub

		Public Sub New(ByVal parentReport As XtraReport, ByVal viewer As DocumentViewer)
			InitializeComponent()
			Me.Viewer = viewer
		End Sub

		Private Sub xrLabel2_PreviewMouseMove(ByVal sender As Object, ByVal e As PreviewMouseEventArgs) Handles xrLabel2.PreviewMouseMove
			e.PreviewControl.Cursor = Cursors.Hand
		End Sub

		Private Sub xrLabel2_PreviewClick(ByVal sender As Object, ByVal e As PreviewMouseEventArgs) Handles xrLabel2.PreviewClick
			Dim productReport As New ProductsReport(Me, Viewer)
			productReport.Parameters("CategoryID").Value = DirectCast(DirectCast(e.Brick.Value, DataRowView).Row("CategoryID"), Integer)
			productReport.Parameters("CategoryName").Value = DirectCast(e.Brick.Value, DataRowView).Row("CategoryName").ToString()
			productReport.CreateDocument()
			Viewer.DocumentSource = productReport
		End Sub

		Private Sub xrLabel2_BeforePrint(ByVal sender As Object, ByVal e As System.Drawing.Printing.PrintEventArgs) Handles xrLabel2.BeforePrint
			DirectCast(sender, XRLabel).Tag = GetCurrentRow()
		End Sub
	End Class
End Namespace

File: CS/ExtendedDrillThroughReport/CategoriesReport.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using DevExpress.XtraReports.UI;
using System.Windows.Forms;
using System.Data;
using DevExpress.XtraPrinting;
using DevExpress.XtraPrinting.Preview;

namespace ExtendedDrillThroughReport {
    public partial class CategoriesReport : DevExpress.XtraReports.UI.XtraReport, IDrillThroughReportHelper {

        #region IDrillThroughReportHelper
        public DocumentViewer Viewer { get; set; }
        public XtraReport ParentReport { get; set; }
        #endregion

        public CategoriesReport() {
            InitializeComponent();
        }

        public CategoriesReport(XtraReport parentReport, DocumentViewer viewer) {
            InitializeComponent();
            Viewer = viewer;
        }

        private void xrLabel2_PreviewMouseMove(object sender, PreviewMouseEventArgs e) {
            e.PreviewControl.Cursor = Cursors.Hand;
        }

        private void xrLabel2_PreviewClick(object sender, PreviewMouseEventArgs e) {
            ProductsReport productReport = new ProductsReport(this, Viewer);
            productReport.Parameters["CategoryID"].Value = (int)((DataRowView)e.Brick.Value).Row["CategoryID"];
            productReport.Parameters["CategoryName"].Value = ((DataRowView)e.Brick.Value).Row["CategoryName"].ToString();
            productReport.CreateDocument();
            Viewer.DocumentSource = productReport;
        }

        private void xrLabel2_BeforePrint(object sender, System.Drawing.Printing.PrintEventArgs e) {
            ((XRLabel)sender).Tag = GetCurrentRow();
        }
    }
}
